tests/unittests/general/bulkparse_test.py

```diff
@@ -84,7 +86,7 @@ def test_three_lines_with_two_rows_should_be_counted_as_three(self):
             b"room1:10.0.0.187:myorg:OTHER:SNMP v2c read profile::\n"
         )
         b = bulkparse.NetboxBulkParser(data)
-        out_data = list(b)
+        itertools.consume(b)
```

Comment:
That failed pretty fast. I know this was my suggestion, but I must have hallucinated it. I might have built such thing myself at some point.  While modules like `itertools` have lots of little utilities to deal with iterators, this wasn't one of them.

You could use `collections.deque` with a maximum length of 0 as a fast way to explicitly exhaust a generator, although it doesn't read as well:
```suggestion
        deque(b, maxlen=0)
```

Reply:
or, you could keep using the original list statement and add a short comment as to why, so people don't read it and scratch their heads...

Reply:
You didn't hallucinate it: https://docs.python.org/3/library/itertools.html#itertools-recipes